docs/testing/README.md

```diff
@@ -0,0 +1,138 @@
+
+Next paragraphs explain how all those different types of testing are used to keep Calypso healthy.
+
+### Server side tests
+
+This test configuration contains unit tests that verify code located in `server` top level folder. 
+
+It supports automatic test discovery. We only need to put a test file into a `test` subfolder, next to the files we want to test.
+
+Tests can be run in 3 different modes:
+```bash
+> # run the entire server suite
+> npm run test-server
+> # run a configuration customized to work with continuous integration 
+> npm run test-server:ci
+> # run tests in watch mode, by default it executes tests only for the modified files 
+> npm run test-client:watch
+```
+
+Those tests are executed on every push on continuous integration (we use CircleCi). This is why all individual tests need to be blazing fast. Please note that network connection is disabled for this configuration.
+
+_Check also how to write [unit components](unit-tests.md)._
+
+### Client side tests
+
+This test configuration unit and component tests that verify code located in `client` top level folder. 
+
+It supports automatic test discovery. We only need to put a test file into a `test` subfolder, next to the files we want to test. 
+
+Tests can be run in 3 different modes:
+```bash
+> # run the entire client suite
+> npm run test-client
+> # run a configuration customized to work with continuous integration
+> npm run test-server:ci
+> # run tests in watch mode, by default it executes tests only for the modified files 
+> npm run test-client:watch
+```
+
+They are executed on every push on continuous integration (CircleCI). This is why all individual tests need to be blazing fast. Please note that network connection is disabled for this configuration.
+
+_Check also how to write [unit components](unit-tests.md) and [component tests](component-tests.md)._
+
+### Integration tests
+
+This test configuration contains integration tests that verify code located in `bin`, `client`, `server` and `test` top level folders. They should test how a group of components or a larger part of business logic works together.
+
+It supports automatic test discovery. We only need to put a test file into a `integration` subfolder, next to the files we want to test.
+
+Tests can be run in 2 different modes:
+```bash
+> # run the entire integration suite
+> npm run test-integration
+> # run a configuration customized to work with continuous integration
+> npm run test-integration:ci
+```
+
+They run daily on continuous integration (CircleCI), because they can use network connection or memory intensive processing and therefore can have longer runtime.
+
+### End-to-end tests
```

Comment:
@lancewillett, @alisterscott and @hoverduck It would be great to provide more details here to encourage people to start writing e2e tests, too. It would be awesome to make it as a requirement when exposing new festures to the users.